### PR TITLE
#53 - dot notation helper

### LINKED ISSUE #53: Dot Notation Helper
Add Support for Dot Notation in JSON Assertions

### PR DESCRIPTION
**Pull request by PiotrFedak**
Created dot notation helper 

This should close #53

### CODE REVIEW

File: src/Helpers/DotNotationHelper.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\BLT\Helpers;
+
+class DotNotationHelper
+{
+    public static function getValueUsingDotNotation(array $array, string $path)
```
Comment (EwelinaSkrzypacz) [inline src/Helpers/DotNotationHelper.php:9]: Missing return type?

File: tests/HelpersTest.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    {
+        $input = "null";
+        $expected = null;
+
+        $actual = NullableHelper::toNullable($input);
+
+        self::assertSame($expected, $actual);
+    }
+
+    public function testToNullableReturnsInputForNonNullString(): void
```
Comment (EwelinaSkrzypacz) [inline tests/HelpersTest.php:24]:
```suggestion
    public function testToNullableReturnsInputForNotEmptyString(): void
```

**Comment by EwelinaSkrzypacz** [comment]
And for me you should separate tests - one helper - one test file

File: src/Helpers/DotNotationHelper.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\BLT\Helpers;
+
+class DotNotationHelper
+{
+    public static function getValueUsingDotNotation(array $array, string $path): mixed
```
Comment (krzysztofrewak) [inline src/Helpers/DotNotationHelper.php:9]:
```suggestion
    public static function getValueUsingDotNotation(array $array, string $path): ?array
```

`mixed` is too wide for this scope. Use nullable type.

File: tests/DotNotationHelperTest.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        $array = [
+            "user" => [
+                "profile" => [
+                    "email" => "user@example.com",
+                    "age" => 25,
+                ],
+            ],
+        ];
+
+        $this->assertEquals("user@example.com", DotNotationHelper::getValueUsingDotNotation($array, "user.profile.email"));
```
Comment (krzysztofrewak) [inline tests/DotNotationHelperTest.php:23]: Why not use Laravel's `[data_get()](https://laravel.com/docs/11.x/helpers#method-data-get)`? You could move `illuminate/support` from dev dependecies to dependencies in `composer.json` and use it directly here.

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: src/Helpers/DotNotationHelper.php (hunk 3 above)
Comment (PiotrFedak) [reply src/Helpers/DotNotationHelper.php:9]: okey, I'm changing it now

File: tests/DotNotationHelperTest.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        $this->assertEquals(25, DotNotationHelper::getValueUsingDotNotation($array, "user.profile.age"));
+        $this->assertNull(DotNotationHelper::getValueUsingDotNotation($array, "user.profile.name"));
+    }
+
+    public function testGetValueUsingDotNotationWithEmptyPath(): void
```
Comment (EwelinaSkrzypacz) [inline tests/DotNotationHelperTest.php:29]:
```suggestion
    public function testReturnsNullWithEmptyPath(): void
```

Comment (EwelinaSkrzypacz) [reply tests/DotNotationHelperTest.php:29]: I think that we can rid of helpers' names in test names.

File: tests/DotNotationHelperTest.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                ],
+            ],
+        ];
+
+        $this->assertEquals("user@example.com", data_get($array, "user.profile.email"));
+        $this->assertEquals(25, data_get($array, "user.profile.age"));
+        $this->assertNull(data_get($array, "user.profile.name"));
+    }
+
+    public function testReturnsNullWithEmptyPath(): void
```
Comment (EwelinaSkrzypacz) [inline tests/DotNotationHelperTest.php:28]: If you change tests' name in one file, please be consistent and change in other files too.

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Comment by PiotrFedak** [comment]
Closed because This won't be necessary. To handle object that using "dot" notation we can use data_get(). So it seems to me that we don't need to have a separate helper here.

**Pull request closed without merging**